src/components/FilterGroup.tsx

```diff
@@ -85,17 +93,61 @@ export function FilterGroup({
       {renderTitle()}
       <CollapsibleSection className={cssClasses.optionsContainer}>
         {searchable && <SearchInput className={cssClasses.searchInput} />}
-        {filterOptions.map(o => {
-          return (
-            <CheckboxOption
-              {...o}
-              key={o.displayName || o.value.toString()}
-              customCssClasses={cssClasses}
-            />
-          );
-        })}
+        <CheckboxOptions
+          filterOptions={filterOptions}
+          showMoreLimit={showMoreLimit}
+          cssClasses={cssClasses} />
         {children}
       </CollapsibleSection>
     </FilterGroupProvider>
   );
 }
+
+function CheckboxOptions({
+  filterOptions,
+  showMoreLimit,
+  cssClasses
+}: {
+  filterOptions: FilterOptionConfig[],
+  showMoreLimit: number,
+  cssClasses: CheckboxCssClasses
+}) {
+  const searchUtilities = useSearchUtilities();
+  const { searchValue } = useFilterGroupContext();
+
+  const shouldRenderOption = (
+    option: FilterOptionConfig
+  ) => {
+    if (isUndefined(option.displayName)) {
```

Comment:
nit: can we just have `option.displayName = option.displayName || option.value.toString()` instead of this check?